osu.Game/Screens/Edit/Compose/Components/BeatDivisorControl.cs

```diff
@@ -220,6 +222,36 @@ protected override bool OnKeyDown(KeyDownEvent e)
             return base.OnKeyDown(e);
         }
 
+        public virtual bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
+        {
+            switch (e.Action)
+            {
+                case GlobalAction.EditorCycleNextBeatSnapDivisor:
+                    cycle(1);
+                    return true;
+
+                case GlobalAction.EditorCyclePreviousBeatSnapDivisor:
+                    cycle(-1);
+                    return true;
```

Comment:
Is there a particular reason this is using the new `cycle()` method rather than `BindableBeatDivisor.{Previous,Next}()` that the chevron buttons on the beat divisor controls are already using?

https://github.com/ppy/osu/blob/15725fb18611925b70ed82396675bd419b848d70/osu.Game/Screens/Edit/Compose/Components/BeatDivisorControl.cs#L103-L113

Reply:
I had a strong feeling something like this already existed.

I think we can combine usage, although we will need to remove the loop-at-extremity to match stable behaviour – do you think that's okay to apply to the buttons too or should it be a `bool` param?